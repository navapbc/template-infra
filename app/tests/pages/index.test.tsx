// test/pages/index.test.js
import { render, screen } from '@testing-library/react'
import { axe } from 'jest-axe'

import Index from '../../src/pages/index'

describe('Index', () => {
  it('should render the heading', () => {
    render(<Index />)

    const heading = screen.getByText(/Welcome/i)

    expect(heading).toBeInTheDocument()
    expect(heading).toMatchSnapshot()
  })

  it('should pass accessibility scan', async () => {
    const { container } = render(<Index />)
    const results = await axe(container)

    expect(results).toHaveNoViolations()
  })
})
