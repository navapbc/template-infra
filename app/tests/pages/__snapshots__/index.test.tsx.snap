// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index should render the heading 1`] = `
<h1
  class="title"
>
  Welcome to your 
  <a
    href="https://github.com/navapbc/template-application-nextjs"
  >
    Next.js Template!
  </a>
</h1>
`;
